### PR TITLE
ES6 Rewrite and 0.4.0 features

### PR DESCRIPTION
**Pull request by novacrazy**
I saw this hadn't been updated in a while and decided to rewrite it using ES6 syntax to make it easier in general. I also added the `clear` method referenced in  #6 and the J. Resig constructors in #10, plus added recent Node and io.js travis support as mentioned in #11 

I did not include the default instance as mentioned in #9 because it's a good way to pollute global state that way.

Other than that, it's basically the same. It uses Babel's UMD (Universal Module Definition) transformer to account for browsers and other environments and even though the transpiled code is a little messy it behaves the same, perhaps even with a few performance improvements with iterations.

### CODE REVIEW

**Comment by gustavohenke** [comment]
That's the exact expression that came into mind when I saw this amazing PR! :clap: :clap: 

Great work. I'll make sure to review this code within the upcoming days.
And thanks a lot for taking the time use ES6 :+1:

**Comment by novacrazy** [comment]
I would have added in eshint in place of jshint, but I don't think it takes too kindly to the es7 strawman proposals like class properties yet.

**Comment by novacrazy** [comment]
Alright, babel-eslint is in. Feel free to add any more rules to it you want. 

Also, I just love working with ES6 and ES7 code. It's so much cleaner and simpler overall. Like replacing the forEach 'loops' for `for..of` loops is just cool. Plus 'native' classes, modules, arrow functions with automatic `this` aliasing and everything else helps.

**Comment by novacrazy** [comment]
I think that's about all I can do with it, actually. Your version seemed pretty optimized from the start. 

I'll let you review it and if you need any help with ES6 stuff or whatever feel free to ask. :)

File: Gruntfile.js [hunk 1]
```diff
@@ -1,55 +1,109 @@
...
+                    optional: [
+                        'spec.undefinedToVoid',
+                        'minification.constantFolding',
+                        'minification.propertyLiterals'
+                    ]
                 },
-                src: "test/spec.js"
+                files:   [{
+                    expand: true,
+                    cwd:    './test/src/',
```
Comment (gustavohenke) [inline Gruntfile.js]: Do we need to transpile the tests? I think they could definitely stay in ES5

Comment (novacrazy) [inline Gruntfile.js]: Fixed.

**Comment by gustavohenke** [comment]
Let me know when you're done, and I'll merge.

I think we should bump to 1.0.0 instead of 0.4.0. The lib has been stable for a good amount of time.
Perhaps publish as a beta and do some tests before releasing the final 1.0.0.

**Comment by novacrazy** [comment]
Alright, I'm done with it. Thanks for taking the time to look over everything, it was fun :)

**Comment by gustavohenke** [comment]
Alright. Anyway, I added you as an collaborator :)
Feel free to make any modifications you want.

**Pull request merged**